calcom/cal.com
# feat: Instant Meeting

- [x] Add "Instant Meeting" to event-types menu
- [x] If a person is available, show a link: "Join Meeting" immediately
- [x] use webhook  for notifications
- [x] have a dynamic link
- [x] i.e.: http://app.cal.com/instant-meeting/6YsEEOIom8HbXrzQOWdo
- [x] require logged in user for the "host"
- [x] track who joins (need logged in [cal.com](http://cal.com/) account)
- [x] trigger a full booking flow (as if someone booked a time in the future)
- [x] add that event to their account (for analytics purposes)
- [x] fastest to join gets the call
- [x] after one person clicked the link, show a dialog: "Someone already joined the meeting. Do you want to join as well?"
- [x] requires team to be on enterprise (Upgrade button)

How to test ?

Instant Meeting v1

Instant Meetings are only available for org team event types.
Only Cal video room is supported.
Booker books the event and a slack message is sent with url to connect and join to meeting link as configured in zapier by team admin/team member.
only team member part of the same team as event type would be able to accept the booking

Create team event type for organization and enable "Instant Booking" switch in event type settings

1) Booker goes to org team event type and sees a bottom banner "Don't want to wait?".
2) Click on "connect now" should directly open BookEventForm with slot start time as current time and duration as default duration.
3) Booker books the event and sees a modal with text "we are connecting you to someone".
4) Now check the DB and get the instant meeting token table and copy the token and go to url http://app.cal.local:3000/connect-and-join?token={token}. 
5) Click on Join Meeting to accept the meeting and redirect to the meeting url

Loom https://www.loom.com/share/f8e31fcc45b04b4abec362fbe94d50ac?sid=3a95a0bd-e136-4886-bae3-1daa95629182

![image](https://github.com/calcom/cal.com/assets/8019099/b04c3baf-eb90-40bf-a5ad-26b884729fda)


https://github.com/calcom/cal.com/assets/8019099/e40cec2d-13ee-4110-9d3c-781481d8b72c
